conda/plugins/subcommands/doctor/health_checks.py

```diff
@@ -175,6 +177,44 @@ def requests_ca_bundle_check(prefix: str, verbose: bool) -> None:
             )
 
 
+def consistent_env_check(prefix: str, verbose: bool) -> None:
+    # get the plugin manager from context
+    pm = context.plugin_manager
+
+    # get prefix data
+    pd = PrefixData(prefix)
+
+    # get virtual packages record
+    virtual_packages = {
+        record.name: record for record in pm.get_virtual_package_records()
+    }
+
+    # collect missing dependency issues in a list
+    issues = []
+
+    # check is dependencies are present
+    for record in pd.iter_records():
+        for dependency in record.depends:
+            match_spec = MatchSpec(dependency)
+            dependency_record = pd.get(
+                match_spec.name, default=virtual_packages.get(match_spec.name, None)
+            )
+            if dependency_record is None:
+                issues.append(f"{match_spec} required by {record.name} not found")
+                continue
+            elif not match_spec.match(dependency_record):
+                issues.append(
+                    f"{record.name}:\n installed dependency '{dependency_record.name}=={dependency_record.version}' does not meet the required dependency constrains '{match_spec}'"
```

Comment:
Knowing where the contrain is coming from would be helpful, maybe adding something like `from {depencency.name}`?
This is particularly helpful since removing the source of the constraint is a possible remediation action.